contracts/test/TimeLockTest.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.9;
+
+import { TimeLock } from '../utils/TimeLock.sol';
+
+contract TimeLockTest is TimeLock {
+    uint256 public num;
+
+    event NumUpdated(uint256 newNum);
+
+    // solhint-disable-next-line no-empty-blocks
+    constructor(uint256 _minimumTimeDelay) TimeLock(_minimumTimeDelay) {}
+
+    function getNum() external view returns (uint256) {
+        return num;
+    }
+
+    function scheduleSetNum(bytes32 _hash, uint256 _eta) external {
```

Comment:
Similar to our governance contract, the `hash` should be derived from the params on-chain, `num` in this case, to better emulate the usage

Reply:
Makes sense, I changed it so the hash is calculated on-chain and made a separate function to test the invalid hash edge case.